src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrRepoDeletionTest.cs

```diff
@@ -0,0 +1,150 @@
+            }
+        };
+
+        File.WriteAllText(InstallerRepoPath / _sourceMappingsRelativePath,
+            JsonSerializer.Serialize(_sourceMappings, _jsonSettings));
+
+        await GitOperations.CommitAll(InstallerRepoPath, "Change source-mappings");
+
+        await UpdateRepoToLastCommit(Constants.InstallerRepoName, InstallerRepoPath);
+
+        expectedFilesFromRepos = new List<LocalPath>
+        {
+            VmrPath / VmrInfo.SourcesDir / "some-file.txt",
+            VmrPath / VmrInfo.SourcesDir / Constants.InstallerRepoName / _sourceMappingsRelativePath,
+            VmrPath / VmrInfo.SourcesDir / Constants.InstallerRepoName / Constants.GetRepoFileName(Constants.InstallerRepoName),
+        };
+
+        expectedFiles = GetExpectedFilesInVmr(
+            VmrPath,
+            new[] { Constants.InstallerRepoName },
+            expectedFilesFromRepos);
+
+        CheckDirectoryContents(VmrPath, expectedFiles);
+
+        var versions = AllVersionsPropsFile.DeserializeFromXml(VmrPath / VmrInfo.GitInfoSourcesDir / AllVersionsPropsFile.FileName);
+        versions.Versions.Keys.Should().BeEquivalentTo(new string[] { "installerGitCommitHash", "installerOutputPackageVersion" });
+
+        var sourceManifest = SourceManifest.FromJson(Info.GetSourceManifestPath());
+        sourceManifest.Repositories.Should().HaveCount(1);
+        sourceManifest.Repositories.First().Path.Should().Be("installer");
+
+        await GitOperations.CheckAllIsCommitted(VmrPath);
+    }
+
+    protected override async Task CopyReposForCurrentTest()
+    {
+        await CopyRepoAndCreateVersionDetails(CurrentTestDirectory, Constants.InstallerRepoName);
+        await CopyRepoAndCreateVersionDetails(CurrentTestDirectory, Constants.ProductRepoName);
+
+        _sourceMappings = new SourceMappingFile
+        {
+            PatchesPath = "src/installer/patches/",
+            SourceMappingsPath = "src/installer/src/SourceBuild/content/source-mappings.json",
+            AdditionalMappings = new List<AdditionalMappingSetting>
+            {
+                new AdditionalMappingSetting
+                {
+                    Source = "src/installer/src/SourceBuild/content/source-mappings.json",
+                    Destination = "src"
+                }
+            },
+            Mappings = new List<SourceMappingSetting>
+            {
+                new SourceMappingSetting
+                {
+                    Name = Constants.InstallerRepoName,
+                    DefaultRemote = InstallerRepoPath,
+                },
+                new SourceMappingSetting
+                {
+                    Name = Constants.ProductRepoName,
+                    DefaultRemote = ProductRepoPath,
+                },
+            }
+        };
+
+        Directory.CreateDirectory(InstallerRepoPath / "src" / "SourceBuild" / "content");
+
+        File.WriteAllText(InstallerRepoPath / _sourceMappingsRelativePath,
```

Comment:
There are usually `*Async` version of these methods so since we are already in an async method, I'd use those where possible